src/components/molecules/EarthEditor/PropertyPane/PropertyField/ColorField/hooks.ts

```diff
@@ -0,0 +1,150 @@
+import { useState, useEffect, useCallback, useRef } from "react";
+import tinycolor, { ColorInput } from "tinycolor2";
+
+export type RGBA = {
+  r: number;
+  g: number;
+  b: number;
+  a: number;
+};
+
+export type Params = {
+  value?: string;
+  onChange?: (value: string | null) => void | undefined;
+};
+
+export default ({ value, onChange }: Params) => {
+  const [colorState, setColor] = useState<string | null>(null);
+  const [rgba, setRgba] = useState<RGBA>(tinycolor(value).toRgb());
+  const [open, setOpen] = useState(false);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const pickerRef = useRef<HTMLDivElement>(null);
+
+  const getHexString = (value?: ColorInput) => {
+    if (!value) return undefined;
+    const color = tinycolor(value);
+    return color.getAlpha() === 1 ? color.toHexString() : color.toHex8String();
+  };
+
+  const handleChange = useCallback(newColor => {
+    const color = getHexString(newColor);
+    if (!color) return;
+    setColor(color);
+    setRgba(newColor);
+  }, []);
+
+  const handleRgbaInput = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      e.preventDefault();
+
+      if (e.target.name === "a") {
+        setRgba({
+          ...rgba,
+          [e.target.name]: Number(e.target.value) / 100,
+        });
+      } else {
+        setRgba({
+          ...rgba,
+          [e.target.name]: e.target.value ? Number(e.target.value) : undefined,
+        });
+      }
+    },
+    [rgba],
```

Comment:
Same as above, don't we need to add `setRgba` to the dependencies list?

Reply:
Same as above